presto-rcfile/src/main/java/com/facebook/presto/rcfile/RcFileReader.java

```diff
@@ -286,11 +287,11 @@ public void close()
         rowGroupPosition = 0;
         rowGroupRowCount = 0;
         currentChunkRowCount = 0;
-        input.close();
-        if (decompressor != null) {
-            decompressor.destroy();
-        }
 
+        try (Closer closer = Closer.create()) {
+            closer.register(input);
+            closer.register(decompressor::destroy);
+        }
```

Comment:
this was my initial version, but with this one we will swallow input.close's exception. If you are fine with that I will update.